utils/rwrecorder.go

```diff
@@ -17,44 +17,44 @@ import (
 	"golang.org/x/net/http/httpguts"
 )
 
-type rwRecorder struct {
+type RWRecorder struct {
 	statusCode int
 	h          http.Header
-	w          io.ReadWriter
+	W          io.ReadWriter
```

Comment:
Alessandro made comments to try and pull out similar logic, so I moved `rwrecorder.go` to `utlls/` to work towards that goal. `W` is exported as it is used in the existing app channel. I agree that ideally this is unexported, but since it was used in the existing app channel, and I wanted to avoid making changes there in this PR to keep the existing logic as is and bring in a side by side external app channel, then the clean up to reduce code between the two app channels has led to this. I can look if this could be prevented, but also wanted to avoid making changes to this file where possible as it is brought in from another Go repo (giving appropriate credits).